cpp/include/raft/spatial/knn/knn.cuh

```diff
@@ -38,65 +43,121 @@ namespace knn {
  *
  * etc...
  *
- * @tparam value_idx
+ * @tparam idx_t
```

Comment:
Not sure, this PR mainly updates the `select_k` below. I only update the template parameter names so they look more alike the rest.

Reply:
@tfeher, yes it's correct because it's performing the same k-select it would perform over a tile of pairwise distances, but it's doing it over potentially multiple k-selected partitions `n_partsxnxk` to reduce them into a single set of size `nxk`.